task/backend/store.go

```diff
@@ -189,54 +186,23 @@ type UpdateTaskRequest struct {
 	options.Options
 }
 
-// UpdateFlux updates the taskupdate to go from updating options to updating a flux string, that now has those updated options in it
+// UpdateFlux updates the TaskUpdate to go from updating options to updating a flux string, that now has those updated options in it
 // It zeros the options in the TaskUpdate.
 func (t *UpdateTaskRequest) UpdateFlux(oldFlux string) error {
-	if t.Script != "" {
-		oldFlux = t.Script
-	}
-	parsedPKG := parser.ParseSource(oldFlux)
-	if ast.Check(parsedPKG) > 0 {
-		return ast.GetError(parsedPKG)
+	if t.Options.IsZero() {
+		return nil
 	}
-	parsed := parsedPKG.Files[0] //TODO: remove this line when flux 0.14 is upgraded into platform
-	// so we don't allocate if we are just changing the status
-	if t.Every != 0 && t.Cron != "" {
-		return errors.New("cannot specify both every and cron")
+	tu := platform.TaskUpdate{
```

Comment:
If I'm understanding this correctly the backend store no long does its own update flux functionality but now relies on the influxdb.TaskUpdate to do it. If thats true then great. My quiestion is why have this function at all? I think it could be removed.

Reply:
It is still in, because the format of the update objects is different between the two packages, the internal update format needs to be beefier than the external facing one, and the zero values are a little different (needed some string pointers) :-(